tests/test_keys.py

```diff
@@ -197,6 +197,29 @@ def test_format_metadata_to_key(self):
 
 
 
+  def test_format_metadata_to_key_non_default_hash_algorithms(self):
+    # Reconfiguring rsakey_dict to conform to KEY_SCHEMA
+    # i.e. {keytype: 'rsa', keyval: {public: pub_key, private: priv_key}}
+    keyid = self.rsakey_dict['keyid']
+    del self.rsakey_dict['keyid']
+
+    rsakey_dict_from_meta, junk = KEYS.format_metadata_to_key(self.rsakey_dict,
+      keyid_hash_algorithms=['sha384'])
+
+    # Check if the format of the object returned by this function corresponds
+    # to RSAKEY_SCHEMA format.
+    self.assertEqual(None,
+        securesystemslib.formats.RSAKEY_SCHEMA.check_match(rsakey_dict_from_meta),
```

Comment:
Should we use `assertTrue()` and [`matches()`](https://github.com/secure-systems-lab/securesystemslib/blob/09cf6d4fe8c5131944a8cf0f109fae7d14ba29b2/securesystemslib/schema.py#L70), rather than `assertEqual()` and [`check_match()`](https://github.com/secure-systems-lab/securesystemslib/blob/09cf6d4fe8c5131944a8cf0f109fae7d14ba29b2/securesystemslib/schema.py#L86), here? 

`matches()` returns a boolean to indicate whether the schema is matched, whereas `check_match()` returns `None` on a match and throws an exception on a mismatch. I don't think we want tests throwing unhandled exceptions and testing for a `None`-value or an `Exception` requires more code.

Reply:
That's a good suggestion. I will add it.